### PR TITLE
[ZH] Add osu file formats

### PR DESCRIPTION
**Pull request by PercyDan54**
Finished rebasing my repo of #4223. I was a noob using git back then.

### CODE REVIEW

**Comment by TicClick** [comment]
cool. does [kj415j45's comment](https://github.com/ppy/osu-wiki/pull/4223#pullrequestreview-511033198) mean the approval?

File: wiki/osu!_File_Formats/Db_(file_format)/zh.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+为简化每个.db文件格式的描述，下列数据类型将被使用。 除非特别指定，否则所有数字类型都以低字节序存储。 整数值，包括字节，都是无符号的。 UTF-8字符以其规范形式存储，优先存储高阶字节。
+
+| 名称 | 字节数 | 描述 |
+| :-- | :-- | :-- |
+| Byte | 1 | 整数 |
+| Short | 2 | 整数 |
+| Int | 4 | 整数 |
+| Long | 8 | 整数 |
+| ULEB128 | 可变 | 整数; 见[此处](https://en.wikipedia.org/wiki/LEB128) |
```
Comment (TicClick) [inline wiki/osu!_File_Formats/Db_(file_format)/zh.md:22]:
```suggestion
| ULEB128 | 可变 | 整数; 见[此处](https://en.wikipedia.org/wiki/LEB128 "Wikipedia") |
```

File: wiki/osu!_File_Formats/Db_(file_format)/zh.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+| Single | 4 | 32位 IEEE 浮点数值 |
+| Double | 8 | 64位 IEEE 浮点数值 |
+| Boolean | 1 | 0x00 为false，其余任何值为true |
+| String | 可变 | 分为三部分：一个单独的字节，可能为 0x00，表示接下来两个部分不存在，或 0x0b（十进制 11），表示下面两个部分存在。若为 0x0b，第二部分将有一个 ULEB128，表示第三部分的字符串字节长度，然后是字符串本身，经 UTF-8 编码。见 [此处](https://en.wikipedia.org/wiki/UTF-8)。 |
```
Comment (TicClick) [inline wiki/osu!_File_Formats/Db_(file_format)/zh.md:26]:
```suggestion
| String | 可变 | 分为三部分：一个单独的字节，可能为 0x00，表示接下来两个部分不存在，或 0x0b（十进制 11），表示下面两个部分存在。若为 0x0b，第二部分将有一个 ULEB128，表示第三部分的字符串字节长度，然后是字符串本身，经 UTF-8 编码。见 [此处](https://en.wikipedia.org/wiki/UTF-8 "Wikipedia")。 |
```

File: wiki/osu!_File_Formats/Db_(file_format)/zh.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+### 格式
+
+一些 osu!.db 专用的数据类型如下所示：
+
+| 名称 | 字节数 | 描述 |
+| :-- | :-- | :-- |
+| Int-Double 对 | 14 | 第一个字节为0x08，跟随一个 Int，然后是0x0d，跟随一个 Double。 这些无关的字节大概是表示这些位置中不同数据类型的标志，但实际上还没有看到其他这样的标志。目前，这种数据类型的用途是未知的。 |
+| Timing Point | 17 | 包含一个代表 BPM 的 Double，和一个代表偏移量（以毫秒为单位）的 Double，和一个布尔值；如果为 false，则该 Timing Point 是继承的。查阅 [.osu（文件格式）](/wiki/osu!_File_Formats/Osu_(file_format)) 以获得更多关于 Timing Point 的信息。 |
+| DateTime | 8 | 一个代表日期和时间的 64 位刻。刻是从 UTC 时间 1 年 1 月 1 日午夜 0 点开始的 100 纳秒间隔量。 详见 [.NET Framework 关于刻的文档](https://docs.microsoft.com/zh-cn/dotnet/api/system.datetime.ticks?view=netframework-4.7.2)。 |
```
Comment (TicClick) [inline wiki/osu!_File_Formats/Db_(file_format)/zh.md:40]:
```suggestion
| DateTime | 8 | 一个代表日期和时间的 64 位刻。刻是从 UTC 时间 1 年 1 月 1 日午夜 0 点开始的 100 纳秒间隔量。 详见 [.NET Framework 关于刻的文档](https://docs.microsoft.com/zh-cn/dotnet/api/system.datetime.ticks?view=netframework-4.7.2 "Microsoft Docs")。 |
```

**Comment by PercyDan54** [comment]
> cool. does [kj415j45's comment](https://github.com/ppy/osu-wiki/pull/4223#pullrequestreview-511033198) mean the approval?

Nope. He means he's busy and can't review now.

**Review by TicClick — CHANGES_REQUESTED** [review]

**Comment by TicClick** [comment]
alright. do you want to wait for someone to pick this up within 2 weeks, or should I consider that these 2 weeks have passed (because #4223 was sitting there for the whole time)

**Comment by PercyDan54** [comment]
> alright. do you want to wait for someone to pick this up within 2 weeks, or should I consider that these 2 weeks have passed (because #4223 was sitting there for the whole time) and merge this right away?

I think I might want to wait for a while.

**Comment by TicClick** [comment]
that's totally alright, it's your feature branch after all

File: wiki/osu!_File_Formats/Db_(file_format)/zh.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+一些osu!.db专用的数据类型如下所示：
+
+| 名称 | 字节数 | 描述 |
+| :-- | :-- | :-- |
+| Int-Double pair | 14 | 第一个字节为0x08，跟随一个Int，然后是0x0d，跟随一个 Double。 这些无关的字节大概是表示这些位置中不同数据类型的标志，但实际上还没有看到其他这样的标志。目前，这种数据类型的用途是未知的。 |
+| Timing Point | 17 | 包含一个代表BPM的Double，和一个代表偏移量（以毫秒为单位）的 Double，和一个布尔值；如果为 false，则该 Timing Point 是继承的。查阅 [.osu（文件格式）](/wiki/osu!_File_Formats/Osu_(file_format)) 以获得更多关于 Timing Point 的信息。 |
+| DateTime | 8 | 一个代表日期和时间的 64 位刻。刻是从 UTC 时间 1 年 1 月 1 日午夜 0 点开始的 100 纳秒间隔量。 详见[.NET Framework 关于刻的文档](https://docs.microsoft.com/zh-cn/dotnet/api/system.datetime.ticks?view=netframework-4.7.2 "Microsoft Docs")。 |
+
+### osu!.db format
```
Comment (Kotoki1337) [inline wiki/osu!_File_Formats/Db_(file_format)/zh.md:42]: 这里是没翻还是要保留原文？

**Review by Kotoki1337 — COMMENTED** [review]
中英之前要不要加下空格format一下

File: wiki/osu!_File_Formats/Db_(file_format)/zh.md (hunk 4 above)
Comment (PercyDan54) [reply wiki/osu!_File_Formats/Db_(file_format)/zh.md]: 漏了

**Review by Kotoki1337 — APPROVED** [review]

**Review by TicClick — APPROVED** [review]

**Pull request merged**